source/d3d12/d3d12_command_queue.cpp

```diff
@@ -168,6 +169,14 @@ void    STDMETHODCALLTYPE D3D12CommandQueue::ExecuteCommandLists(UINT NumCommand
 	// Synchronize access to this command queue while events are invoked and the immediate command list may be accessed
 	std::unique_lock<std::recursive_mutex> lock(_mutex);
 
+#if RESHADE_ADDON
+	// Emit RENDERSUBMIT_START marker at the beginning of a batch submission
```

Comment:
This is not correct: Applications do submit multiple times per frame, and also to multiple different command queues. The `RENDERSUBMIT_START/END` markers however are supposed to be called only once per frame, before all the submissions and after all the submissions. This can't easily be mapped to just hooking `ExecuteCommandLists`.